core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -157,18 +157,57 @@ __global__ static void cuda_parallel_launch_constant_or_global_memory(
   driver();
 }
 
-template <class DriverType, unsigned int maxTperB, unsigned int minBperSM>
-__global__
-__launch_bounds__(maxTperB, minBperSM) static void cuda_parallel_launch_constant_or_global_memory(
-    const DriverType* driver_ptr) {
-  const DriverType& driver =
-      driver_ptr != nullptr
-          ? *driver_ptr
-          : *((const DriverType*)kokkos_impl_cuda_constant_memory_buffer);
+//==============================================================================
+// <editor-fold desc="Some helper functions for launch code readability"> {{{1
 
-  driver();
+inline bool is_empty_launch(dim3 const& grid, dim3 const& block) {
+  return (grid.x == 0) || ((block.x * block.y * block.z) == 0);
+}
+
+inline void check_shmem_request(CudaInternal const* cuda_instance, int shmem) {
+  if (cuda_instance->m_maxShmemPerBlock < shmem) {
+    Kokkos::Impl::throw_runtime_exception(
+        std::string("CudaParallelLaunch (graph node) FAILED: shared memory"
```

Comment:
```suggestion
        std::string("CudaParallelLaunch (or graph node creation) FAILED: shared memory"
```

Probably can't just accept this though because we'd need to rerun clang-format

Reply:
Resolved by 9925f16f055f39372a669e1fd583c4d390ee940b